flutter/news/test/features/frontpage/presentation/bloc/articles_cubit_test.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:news/core/error/failures.dart';
+import 'package:news/core/result.dart';
+import 'package:news/features/frontpage/data/repositories/articles_repository.dart';
+import 'package:news/features/frontpage/domain/entities/article.dart';
+import 'package:news/features/frontpage/domain/entities/source.dart';
+import 'package:news/features/frontpage/presentation/bloc/articles_cubit.dart';
+
+import '../../domain/usecases/get_everything_about_test.mocks.dart';
+
+
+@GenerateMocks([ArticlesRepository])
+void main() {
+  late ArticlesRepository repository;
+  late List<Article> topHeadlines;
+
+  setUp(() {
+    repository = MockArticlesRepository();
+    topHeadlines = [
+      const Article(
+          source: Source(id: "id", name: "name"),
+          author: "author",
+          title: "title",
+          description: "description",
+          url: "url",
+          urlToImage: "urlToImage",
+          publishedAt: "publishedAt",
+          content: "content")
+    ];
+  });
+
+  blocTest<ArticlesCubit, ArticlesState>(
+    'GIVEN topHeadlines is requested WHEN response is successful '
+    'THEN emits [TopHeadlinesLoading, TopHeadlinesLoaded]',
+    build: () => ArticlesCubit(repository: repository),
+    act: (cubit) {
+      when(repository.topHeadlines())
+          .thenAnswer((_) async => Result.success(topHeadlines));
```

Comment:
I have a question: should the stubbing  be part of the `build` instead of the `act`? 

Maybe I'm a bit biased by the `Given` / `When` / `Then` of TDD. 
It seems natural to match them into the `build` / `act` / `expect` of the blocTest

Reply:
@gbasile you are right, it should be part of the `arrange` or `build` section

Reply:
updated on 11abdd537712cd30e5bd412e1657e6fb5f8222ab